### PR TITLE
Implement permissions for topic viewing

### PR DESCRIPTION
**Pull request by ashatna**
Item.urls - add urls
Implement get, post methods for Item views
Modify mentor urls
Mentor views - fix bug in get_students and get mentors methods
Modify is_student method into topic_student_permissions method
UI TopicList container - add components(ItemLink, MentorsChip, TopicItemList, TopicView)
Add permissions for viewing Topic on UI component
TopicLink renamed from TopicItem
Modify TopicDialog, TopicService, TopicList components
Modify containers/index.js
Modify mainRouter
Write tests for Item view
Add test to Mentor.testview, Topic.testview
Modify Topic.urls
Add delete method to Topic views

## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LEAR-38)

## Code reviewers

- [ ] @ashatna
- [ ] @ChristinaPastushchak
- [x] @dotiurin
- [ ] @iryn34
- [x] @tarassito
- [ ] @VasilHamulyak
- [x] @VitalyiKhlivnyi
- [ ] @YurijH

### Second Level Review

- [ ] @ushevtsiv
- [x] @lhalam
- [ ] @ssOleg

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/item/views.py [hunk 1]
```diff
@@ -7,10 +7,92 @@
...
+
+        if topic_id:
+            topic = Topic.get_by_id(topic_id)
+            if not topic:
+                return RESPONSE_404_OBJECT_NOT_FOUND
+            items = topic.item_set.all()
+            data = {'items': [item.to_dict() for item in items]}
+            return JsonResponse(data, status=200)
+
+    def post(self, request, curriculum_id, topic_id):
```
Comment (tarassito) [inline eventually/item/views.py:59]: тобі потрібен curriculum_id?

File: eventually/static/src/containers/topicList/ItemLink.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import Avatar from 'material-ui/Avatar';
+import LibraryBooks from 'material-ui/svg-icons/av/library-books';
+import Code from 'material-ui/svg-icons/action/code';
+import Group from 'material-ui/svg-icons/social/group';
+import FlatButton from 'material-ui/FlatButton';
+import {Card, CardHeader, CardText, CardActions} from 'material-ui/Card';
+import {blue500, yellow600, lime500} from 'material-ui/styles/colors';
+
```
Comment (tarassito) [inline eventually/static/src/containers/topicList/ItemLink.js:9]: add blank line after import(should be 2)

File: eventually/static/src/containers/topicList/TopicItemList.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {ItemLink, getItemListService} from 'src/containers';
+
```
Comment (tarassito) [inline eventually/static/src/containers/topicList/TopicItemList.js:3]: add blank line

File: eventually/static/src/containers/topicList/TopicItemList.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        this.setState({isModalOpen: false});
+    };
+
+    handleAnswerChange = (event, value) => {
+        this.setState({
+            answer: value
+        });
+    };
+
+    handleClick = (id) => {
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/topicList/TopicItemList.js]: unnecessary brackets

File: eventually/static/src/containers/topicList/TopicLink.js [hunk 5]
```diff
@@ -16,33 +18,35 @@ const cardHeaderStyle= {
...
-            isStudent: false,
+            isRequestedStudent: false,
+            isApprovedStudent: false,
             leaveDialogOpen:false,
             leaveTopic: false
         };
     }
 
-    cangeExp = (newExpandedState) => {
+    changeExp = (newExpandedState) => {
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/topicList/TopicLink.js]: unnecessary brackets

File: eventually/static/src/containers/topicList/TopicServices.js [hunk 6]
```diff
@@ -15,19 +14,43 @@ const postTopicAssignService = data => {
     return axios.post(mentorPath, data);
 };
 
-const getTopicStudentsService = (topicId) =>{
+const getTopicStudentsService = (topicId) => {
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/topicList/TopicServices.js]: unnecessary brackets

File: eventually/static/src/containers/topicList/TopicServices.js [hunk 7]
```diff
@@ -15,19 +14,43 @@ const postTopicAssignService = data => {
...
+    let url = curriculumPath + curriculumId + '/topics/' + topicId + '/';
+    return axios.delete(url);
+};
+
+const getItemListService = (curriculumId, topicId) => {
+    let url = curriculumPath + curriculumId + '/topics/' + topicId + '/items/';
+    return axios.get(url);
+};
+
 const deleteMenteeService = (topicId) => {
```
Comment (VitalyiKhlivnyi) [inline eventually/static/src/containers/topicList/TopicServices.js]: unnecessary brackets

File: eventually/topic/views.py [hunk 8]
```diff
@@ -106,6 +109,33 @@ def put(self, request, curriculum_id=None, topic_id=None):
                  400, 403 or 404 failed status code.
         :rtype: `HttpResponse object."""
 
+    def delete(self, request, curriculum_id, topic_id):    # pylint: disable=unused-argument
```
Comment (tarassito) [inline eventually/topic/views.py:112]: curriculum_id ???

File: eventually/item/views.py (hunk 1 above)
Comment (ashatna) [reply eventually/item/views.py:59]: він його отримує з урли, тому має бути прописаний тут

File: eventually/static/src/containers/topicList/ItemLink.js (hunk 2 above)
Comment (ashatna) [reply eventually/static/src/containers/topicList/ItemLink.js:9]: done!

File: eventually/static/src/containers/topicList/TopicItemList.js (hunk 3 above)
Comment (ashatna) [reply eventually/static/src/containers/topicList/TopicItemList.js:3]: done

File: eventually/static/src/containers/topicList/TopicItemList.js (hunk 4 above)
Comment (ashatna) [reply eventually/static/src/containers/topicList/TopicItemList.js]: done!

File: eventually/static/src/containers/topicList/TopicLink.js (hunk 5 above)
Comment (ashatna) [reply eventually/static/src/containers/topicList/TopicLink.js]: done

File: eventually/static/src/containers/topicList/TopicServices.js (hunk 6 above)
Comment (ashatna) [reply eventually/static/src/containers/topicList/TopicServices.js]: done

File: eventually/static/src/containers/topicList/TopicServices.js (hunk 7 above)
Comment (ashatna) [reply eventually/static/src/containers/topicList/TopicServices.js]: done

File: eventually/topic/views.py (hunk 8 above)
Comment (ashatna) [reply eventually/topic/views.py:112]: i need it here )

File: eventually/item/views.py [hunk 9]
```diff
@@ -7,10 +7,92 @@
...
+                                  RESPONSE_404_OBJECT_NOT_FOUND)
+from .models import Item
 
 
 class ItemView(View):
     """
     Item view that handles GET, POST, PUT, DELETE requests and provides appropriate
     operations with item model.
     """
+    def get(self, request, curriculum_id, item_id=None, topic_id=None,):
```
Comment (dotiurin) [inline eventually/item/views.py]: "," in the end of arguments not needed

**Review by dotiurin — DISMISSED** [review]

File: eventually/item/views.py (hunk 9 above)
Comment (ashatna) [reply eventually/item/views.py]: done!

**Review by lhalam — APPROVED** [review]

**Pull request merged**